src/main/metadata.ts

```diff
@@ -0,0 +1,54 @@
+const ROW_ID_KEY: string = "rowId";
+
+/**
+ * Retrieves the metadata object for a given range. If the metadata object does not exist,
+ * this function will set the row id key so the metadata object is created and a reference
+ * can be held
+ * 
+ * @param range Range representing an event row
+ * @returns reference to the Metadata object for this row
+ */
+export function getMetadata(range: Range): Metadata {
+    let metadataList: Metadata[] = range.getDeveloperMetadata();
+
+    if (metadataList.length === 0) {
+        // Add the row id key so the metadata object is created
+        range.addDeveloperMetadata(ROW_ID_KEY);
+        metadataList = range.getDeveloperMetadata();
+    }
+
+    return metadataList[0];
+}
+
+/**
+ * Retrieves the unique id for a Row. The unique id is stored in the value of the metadata.
+ * If the unique id has not been set, null will be returned
+ * 
+ * @param row the Row to retrieve the unique id for
+ * @returns unique id for the row; null if one has not been set
+ */
+export function getRowUniqueId(row: Row): string | null {
```

Comment:
as far as possible, we should try to avoid returning null - it's just waiting for a NPE down the road. i'd rather throw an exception immediately here and we can identify the source of the issue in a controlled manner, vs the NPE popping up somewhere else and it may take longer to debug why

Reply:
i see it's used for rowHasUniqueId, but that could just use row.metadata.getValue() !== null it seems.
i don't want to block this PR since i'll be gone over the weekend, so you can feel free to address this in the next PR or get someone else's approval

Reply:
I'll address this in this PR by throwing an exception. Will get a review from someone else